pkg/instancetype/instancetype.go

```diff
@@ -734,12 +749,26 @@ func (m *InstancetypeMethods) InferDefaultInstancetype(vm *virtv1.VirtualMachine
 
 func (m *InstancetypeMethods) InferDefaultPreference(vm *virtv1.VirtualMachine) (*virtv1.PreferenceMatcher, error) {
 	if vm.Spec.Preference == nil || vm.Spec.Preference.InferFromVolume == "" {
-		return nil, nil
+		return vm.Spec.Preference, nil
 	}
 	defaultName, defaultKind, err := m.inferDefaultsFromVolumes(vm, vm.Spec.Preference.InferFromVolume, apiinstancetype.DefaultPreferenceLabel, apiinstancetype.DefaultPreferenceKindLabel)
+
+	// Ignore certain errors if corresponding policy is set
+	var ignoreableInferenceErr *IgnoreableInferenceError
+	if errors.As(err, &ignoreableInferenceErr) {
+		failurePolicy := vm.Spec.Preference.InferFromVolumeFailurePolicy
+		if failurePolicy != nil && *failurePolicy == virtv1.IgnoreInferFromVolumeFailure {
+			// Lazy logger construction
+			logger := func() *log.FilteredLogger { return log.Log.Object(vm) }
```

Comment:
This is the expected behavior, without unknown side-effects, therefore I think logging it is redundant.

Reply:
Same as above.